drivers/i2c/i2c_renesas_ra_sci.c

```diff
@@ -0,0 +1,739 @@
+/*
+ * Copyright (c) 2025 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra_i2c_sci
+
+#define MDDR_DISABLE 256
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+#include "r_sci_i2c.h"
+#ifdef CONFIG_I2C_RENESAS_RA_SCI_DTC
+#include "r_dtc.h"
+#endif
+
+#include <soc.h>
+
+LOG_MODULE_REGISTER(renesas_ra_i2c_sci, CONFIG_I2C_LOG_LEVEL);
+
+#define I2C_MAX_MSG_LEN (1 << (sizeof(uint8_t) * 8))
+
+struct sci_i2c_config {
+	void (*irq_config_func)(const struct device *dev);
+	const struct pinctrl_dev_config *pcfg;
+	uint16_t sda_output_delay;
+};
+
+struct sci_i2c_data {
+	sci_i2c_instance_ctrl_t ctrl;
+	i2c_master_cfg_t i2c_config;
+	sci_i2c_extended_cfg_t ext_cfg;
+	struct k_sem bus_lock;
+	struct k_sem complete_sem;
+	i2c_master_event_t event;
+	uint32_t dev_config;
+
+#ifdef CONFIG_I2C_CALLBACK
+	uint16_t addr;
+	uint32_t msg_idx;
+	struct i2c_msg *msgs;
+	uint32_t num_msgs;
+	i2c_callback_t cb;
+	void *p_context;
+#endif /* CONFIG_I2C_CALLBACK */
+
+#ifdef CONFIG_I2C_RENESAS_RA_SCI_DTC
+	/* RX */
+	transfer_instance_t rx_transfer;
+
+	transfer_info_t rx_transfer_info DTC_TRANSFER_INFO_ALIGNMENT;
+
+	transfer_cfg_t rx_transfer_cfg;
+	dtc_instance_ctrl_t rx_transfer_ctrl;
+	dtc_extended_cfg_t rx_transfer_cfg_extend;
+
+	/* TX */
+	transfer_instance_t tx_transfer;
+
+	transfer_info_t tx_transfer_info DTC_TRANSFER_INFO_ALIGNMENT;
+
+	transfer_cfg_t tx_transfer_cfg;
+	dtc_instance_ctrl_t tx_transfer_ctrl;
+	dtc_extended_cfg_t tx_transfer_cfg_extend;
+#endif /* CONFIG_I2C_RENESAS_RA_SCI_DTC */
+};
+
+static void calc_sci_iic_clock_setting(const struct device *dev, const uint32_t fsp_i2c_rate,
+					 sci_i2c_clock_settings_t *clk_cfg);
+
+/* FSP interruption handlers. */
+void sci_i2c_txi_isr(void);
+void sci_i2c_tei_isr(void);
+void sci_i2c_rxi_isr(void);
```

Comment:
Doesn't FSP expose it anyhow? Could this be fixed there maybe?

What's the supposed way of using these, should FSP itself configure the IRQs, setting up these handlers?